qa/tasks/fwd_scrub.py

```diff
@@ -0,0 +1,222 @@
+"""
+Thrash mds by simulating failures
+"""
+import logging
+import contextlib
+import threading
+
+from gevent import sleep, killall, joinall, GreenletExit
+from gevent.greenlet import Greenlet
+from gevent.event import Event
+from teuthology import misc as teuthology
+from teuthology import contextutil
+from teuthology.orchestra.run import CommandFailedError
+
+from tasks import ceph_manager
+from tasks.cephfs.filesystem import MDSCluster, Filesystem
+from tasks.thrasher import Thrasher
+
+log = logging.getLogger(__name__)
+
+
+class MDSRankScrubber(Thrasher, Greenlet):
+    def __init__(self, fs, mds_rank, scrub_timeout=300):
+        super(MDSRankScrubber, self).__init__()
+        self.logger = log.getChild('fs.[{f}]'.format(f=fs.name))
+        self.fs = fs
+        self.mds_rank = mds_rank
+        self.scrub_timeout = scrub_timeout
+
+    def _run(self):
+        try:
+            self.do_scrub()
+        except Exception as e:
+            self.set_thrasher_exception(e)
+            self.logger.exception("exception:")
+            # allow successful completion so gevent doesn't see an exception...
+
+    def _wait_until_scrub_complete(self, path="/", recursive=True):
+        recopt = ["recursive", "force"] if recursive else ["force"]
+        out_json = self.fs.rank_tell(["scrub", "start", path] + recopt,
+                                     rank=self.mds_rank)
+        assert out_json is not None
+
+        tag = out_json['scrub_tag']
+
+        assert tag is not None
+        assert out_json['return_code'] == 0
+        assert out_json['mode'] == 'asynchronous'
+
+        # time out after scrub_timeout seconds and assume as done
+        with contextutil.safe_while(sleep=30, tries=self.scrub_timeout//30) as proceed:
+            while proceed():
+                try:
+                    out_json = self.fs.rank_tell(["scrub", "status"],
+                                                 rank=self.mds_rank)
+                    assert out_json is not None
+                    if out_json['status'] == "no active scrubs running":
+                        break
```

Comment:
Add info message that the scrub appears to have completed.

Reply:
done